samples/iolib/src/main/cpp/player/SampleBuffer.cpp

```diff
@@ -74,20 +74,20 @@ void resampleData(const ResampleBlock& input, ResampleBlock* output, int numChan
     float *outputBuffer = new float[numOutFrames];    // multi-channel buffer to be filled
     output->mBuffer = outputBuffer;
 
-    int numOutputFrames = 0;
-    int inputFramesLeft = input.mNumFrames;
-    while (inputFramesLeft > 0) {
+    int numOutputSamples = 0;
+    int inputSamplesLeft = input.mNumSamples;
+    while (inputSamplesLeft > 0) {
         if(resampler->isWriteNeeded()) {
             resampler->writeNextFrame(inputBuffer);
             inputBuffer += numChannels;
-            inputFramesLeft--;
+            inputSamplesLeft -= numChannels;
         } else {
             resampler->readNextFrame(outputBuffer);
             outputBuffer += numChannels;
-            numOutputFrames++;
+            numOutputSamples += numChannels;
         }
     }
-    output->mNumFrames = numOutputFrames;
+    output->mNumSamples = numOutputSamples;
```

Comment:
There may be uninitialized values at the end of the float[] array.
That could cause a problem if we interpolate past the end when playing the waveform.
To be safe we should zero them out here or use a zero initializer on line 74.

Reply:
Actually the sample object may not think there is data past the numSamples.
So it might be OK. Just a little scary.

Reply:
I think this is fine because consumer gets a pointer to a buffer and a size.